agent/consul/state/memdb_test.go

```diff
@@ -0,0 +1,95 @@
+package state
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-memdb"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+	"testing"
+	"time"
+)
+
+func testValidSchema() *memdb.DBSchema {
+	return &memdb.DBSchema{
+		Tables: map[string]*memdb.TableSchema{
+			"main": {
+				Name: "main",
+				Indexes: map[string]*memdb.IndexSchema{
+					"id": {
+						Name:    "id",
+						Unique:  true,
+						Indexer: &memdb.StringFieldIndex{Field: "ID"},
+					},
+					"foo": {
+						Name:    "foo",
+						Indexer: &memdb.StringFieldIndex{Field: "Foo"},
+					},
+				},
+			},
+		},
+	}
+}
+
+type TestObject struct {
+	ID  string
+	Foo string
+}
+
+func Test_txn_Commit(t *testing.T) {
```

Comment:
Would you mind adding a comment here to explain what this test is exercising? If I understand correctly:

- `ch1` is used to trigger a read when the writer publishes its events.
- `ch2` is used to block the publish method from returning until the read has happened.

Before your fix, the test would fail because `ch2` being blocked would ensure the read happened before the transaction was committed. Now, we commit the transaction first. Is that about right?